pkg/controller/acmeorders/sync.go

```diff
@@ -612,13 +635,15 @@ func (c *controller) fetchCertificateData(ctx context.Context, cl acmecl.Interfa
 		return nil
 	}
 
-	// If the Order state has actually changed and we've not observed it,
-	// update the order status and let the change in the resource trigger
-	// a resync
+	return c.fetchCertificateDataWithOrder(ctx, cl, *acmeOrder, o)
+}
+
+func (c *controller) fetchCertificateDataWithOrder(ctx context.Context, cl acmecl.Interface, acmeOrder acmeapi.Order, o *cmacme.Order) error {
```

Comment:
We might at some point want to refactor these helper functions, but it would be a lot of work and I am not sure it's worth doing it right now.